pandas/core/sparse/frame.py

```diff
@@ -138,7 +142,10 @@ def _init_dict(self, data, index, columns, dtype=None):
             columns = _ensure_index(columns)
             data = {k: v for k, v in compat.iteritems(data) if k in columns}
         else:
-            columns = Index(com._try_sort(list(data.keys())))
+            keys = list(data.keys())
```

Comment:
try pushing this entirely to ``_try_sort``, see if anything breaks as this used in a couple of places and will provide some 3.6 consistency.

Reply:
``_try_sort`` is used for sorting lists in a couple of places.

I don't think it's a good idea to have a sort function that doesn't sort on ordered dict keys, but sorts on ordered lists. 

Maybe a differently named function, e.g. ``_dict_keys_to_ordered_list``?

Reply:
ok with renaming this, but do try to change all of the existing functions to use it.
You are creating an inconsistency by making dict inputs follow insertion order, rather have *all* instances fixed.

Reply:
I've changed this for all places where ``_try_sort`` is used to sort  list of dict keys.